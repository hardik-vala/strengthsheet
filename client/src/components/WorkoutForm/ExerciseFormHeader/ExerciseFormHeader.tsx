import { View } from "react-native";
import { Text } from "react-native-paper";
import { Exercise } from "../../../models/Workout/Core";
import { styles } from "../style";

export interface ExerciseFormHeaderProps {
  exercise: Exercise;
}

export function ExerciseFormHeader({ exercise }: ExerciseFormHeaderProps) {
  return (
    <View style={styles.exerciseFormShelfContainer}>
      <View style={styles.exerciseFormShelfSetContainer}>
        <Text variant="bodyMedium" style={styles.exerciseFormShelfText}>
          Set
        </Text>
      </View>
      <View
        style={{
          flexDirection: "row",
        }}
      >
        {exercise.measures.map((measure) => (
          <Text
            key={measure.key}
            variant="bodyMedium"
            style={styles.exerciseFormShelfText}
          >
            {measure.displayName}
          </Text>
        ))}
      </View>
    </View>
  );
}
