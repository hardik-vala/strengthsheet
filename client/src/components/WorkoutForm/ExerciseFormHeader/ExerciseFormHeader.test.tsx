import { render, screen } from "@testing-library/react-native";
import React from "react";
import {
  ExerciseUnit
} from "../../../models/Workout/Core";
import { ExerciseFormHeader } from "./ExerciseFormHeader";

const TEST_EXERCISE = {
  key: "test_exercise",
  displayName: "Test Exercise",
  measures: [
    { key: "lb", displayName: "lb", unit: ExerciseUnit.lb },
    {
      key: "test_measure",
      displayName: "reps",
      unit: ExerciseUnit.reps,
    },
  ],
};

describe("ExerciseFormHeader", () => {
  it("renders correctly", () => {
    render(<ExerciseFormHeader exercise={TEST_EXERCISE} />);

    expect(screen).toMatchSnapshot();
  });
});
