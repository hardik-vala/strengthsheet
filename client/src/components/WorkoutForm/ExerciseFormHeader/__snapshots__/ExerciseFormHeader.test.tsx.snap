// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseFormHeader renders correctly 1`] = `
<View
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingVertical": 5,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 50,
      }
    }
  >
    <Text
      style={
        [
          {
            "textAlign": "left",
          },
          {
            "color": "rgba(28, 27, 31, 1)",
            "writingDirection": "ltr",
          },
          [
            {
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "400",
              "letterSpacing": 0.25,
              "lineHeight": 20,
            },
            {
              "color": "gray",
              "fontWeight": "bold",
              "textAlign": "center",
            },
          ],
        ]
      }
    >
      Set
    </Text>
  </View>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "marginRight": 15,
          "textAlign": "center",
          "width": 75,
        }
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
              {
                "color": "gray",
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ],
          ]
        }
      >
        lb
      </Text>
    </View>
    <View
      style={
        {
          "marginRight": 15,
          "textAlign": "center",
          "width": 75,
        }
      }
    >
      <Text
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontFamily": "System",
                "fontSize": 14,
                "fontWeight": "400",
                "letterSpacing": 0.25,
                "lineHeight": 20,
              },
              {
                "color": "gray",
                "fontWeight": "bold",
                "textAlign": "center",
              },
            ],
          ]
        }
      >
        reps
      </Text>
    </View>
  </View>
</View>
`;
